src/rdata/rfc1035.rs

```diff
@@ -2121,17 +2186,46 @@ impl<Builder: OctetsBuilder + EmptyBuilder> TxtBuilder<Builder> {
 
 #[cfg(feature = "bytes")]
 impl TxtBuilder<BytesMut> {
+    /// Creates a new, empty TXT builder using `BytesMut`.
     pub fn new_bytes() -> Self {
         Self::new()
     }
 }
 
 impl<Builder: OctetsBuilder + AsRef<[u8]> + AsMut<[u8]>> TxtBuilder<Builder> {
-    fn builder_append_slice(&mut self, slice: &[u8]) -> Result<(), ShortBuf> {
-        self.builder.append_slice(slice).map_err(Into::into)
+    /// Tries appending a slice.
+    ///
+    /// Errors out of either appending the slice would result in exceeding the
+    /// record data length limit or the underlying builder runs out of space.
+    fn builder_append_slice(
+        &mut self, slice: &[u8]
+    ) -> Result<(), TxtAppendError> {
+        LongRecordData::check_append_len(
+            self.builder.as_ref().len(), slice.len()
+        )?;
+        self.builder.append_slice(slice)?;
+        Ok(())
     }
 
-    pub fn append_slice(&mut self, mut slice: &[u8]) -> Result<(), ShortBuf> {
+    /// Appends a slice to the builder.
+    ///
+    /// The method breaks up the slice into individual octets strings if
+    /// necessary. If a previous call has started a new octets string, it
+    /// fills this one up first before creating a new one. Thus, be only
```

Comment:
I'm not sure if this is what you mean but I find that "be only using this method" sounds strange to me.